src/shower.cpp

```diff
@@ -149,4 +149,36 @@ void Shower::publish_values() {
     Mqtt::publish(F("shower_data"), doc.as<JsonObject>());
 }
 
+void Shower::send_MQTT_discovery_config() {
+    
+    if (mqtt_discovery_config_send_) {
+        //nothing to do
+        return;
+    }
+
+    //send the config depending on the MQTT format used
+    if (Mqtt::mqtt_format() == Mqtt::Format::HA) {
+        StaticJsonDocument<EMSESP_MAX_JSON_SIZE_HA_CONFIG> doc;
+        doc["name"]    = FJSON("Shower Data");
+        doc["uniq_id"] = FJSON("shower_data");
+        doc["~"]       = FJSON("ems-esp");
```

Comment:
This should be `doc["~"]  = Mqtt::base();`to work with changed configurations.

Reply:
Branch is updated